filter/src/main/java/gov/cms/ab2d/filter/ExplanationOfBenefitTrimmerR4.java

```diff
@@ -2,108 +2,201 @@
+ *     . implicitRules (inherited)
+ *     . patient
+ *     . provider
+ *     . facility
+ *     . careTeam
+ *     . diagnosis (new in R4 so don't use - onAdmission)
+ *     . procedure (new in R4 so don't use - type, udi, udi)
+ *     . billablePeriod
+ *     . item (some of the data)
+ *     . status
+ *
+ *  Items not kept:
+ *     . extension (inherited)
+ *     . modifierExtension (new for R4, inherited)
+ *     . patientTarget
+ *     . created
+ *     . enterer
+ *     . entererTarget
+ *     . insurer
+ *     . insurerTarget
+ *     . provider
+ *     . providerTarget
+ *     . referral
+ *     . referralTarget
+ *     . facilityTarget
+ *     . claim
+ *     . claimTarget
+ *     . claimResponse
+ *     . claimResponseTarget
+ *     . outcome
+ *     . disposition
+ *     . related
+ *     . prescription
+ *     . prescriptionTarget
+ *     . originalPrescription
+ *     . originalPrescriptionTarget
+ *     . payee
+ *     . precedence
+ *     . insurance
+ *     . accident
+ *     . supportingInfo (was information in STU3)
+ *     . addItem
+ *     . payment
+ *     . form
+ *     . contained
+ *     . processNote
+ *     . benefitBalance
+ *     . priority (new for R4)
+ *     . total (new for R4)
+ *     . use (new for R4)
+ *     . fundsReserveRequested (new for R4)
+ *     . fundsReserve (new for R4)
+ *     . preAuthRef (new for R4)
+ *     . preAuthRefPeriod (new for R4)
+ *     . formCode (new for R4)
+ *     . benefitPeriod (new for R4)
+ *     . adjucation (new for R4)
+ *
+ *  For item elements:
+ *
+ *  Keep:
+ *     . sequence
+ *     . careTeamSequence
+ *     . productOrService
+ *     . serviced
+ *     . location
+ *     . quantity
+ *     . extension
+ *
+ *  Items not kept:
+ *     . diagnosisSequence
+ *     . procedureSequence
+ *     . informationSequence
+ *     . extension
+ *     . revenue
+ *     . category
+ *     . modifier
+ *     . programCode
+ *     . unitPrice
+ *     . factor
+ *     . net
+ *     . udi
+ *     . udiTarget
+ *     . bodySite
+ *     . subSite
+ *     . encounter
+ *     . encounterTarget
+ *     . noteNumber
+ *     . adjudication
+ *     . detail
+ *     . modifierExtension
+ *
  */
 public class ExplanationOfBenefitTrimmerR4 {
-
     /**
      * Pass in an ExplanationOfBenefit, return the copy without the data
      *
-     * @param b - the original ExplanationOfBenefit
+     * @param resource - the original ExplanationOfBenefit
      * @return the cleaned up copy
      */
-    public static IBaseResource getBenefit(IBaseResource b) {
-        ExplanationOfBenefit benefit = (ExplanationOfBenefit) b;
+    public static IBaseResource getBenefit(IBaseResource resource) {
+        ExplanationOfBenefit benefit = (ExplanationOfBenefit) resource;
         if (benefit == null) {
             return null;
         }
         // Copy it so we don't destroy the original
-        ExplanationOfBenefit newBenefit = benefit.copy();
+        ExplanationOfBenefit newBenefit = copyData(benefit);
```

Comment:
Why the copy in the first place? It isn't like we need the original for anything. Just seems like an opportunity to miss something and create a bug.

Reply:
I'd rather miss adding data than miss removing it. This way you actively copy data that you want to opt in. I should really change it for the STU3 Trimmer.